src/dev/generate_release_note.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { resolve } from 'path';
+import { readFileSync, writeFileSync, Dirent, renameSync, rm } from 'fs';
+import { load as loadYaml } from 'js-yaml';
+import { mkdir, readdir } from 'fs/promises';
+import { version as pkgVersion } from '../../package.json';
+import {
+  validateFragment,
+  getCurrentDateFormatted,
+  Changelog,
+  SECTION_MAPPING,
+  fragmentDirPath,
+  fragmentTempDirPath,
+  SectionKey,
+  releaseNotesDirPath,
+  filePath,
+} from './generate_release_note_helper';
+
+// Function to add content after the 'Unreleased' section in the changelog
+function addContentAfterUnreleased(path: string, newContent: string): void {
+  let fileContent = readFileSync(path, 'utf8');
+  const targetString = '## [Unreleased]';
+  const targetIndex = fileContent.indexOf(targetString);
+
+  if (targetIndex !== -1) {
+    const endOfLineIndex = fileContent.indexOf('\n', targetIndex);
+
+    if (endOfLineIndex !== -1) {
+      fileContent =
+        fileContent.slice(0, endOfLineIndex + 1) +
+        '\n' +
+        newContent +
+        fileContent.slice(endOfLineIndex + 1);
+    } else {
+      throw new Error('End of line for "Unreleased" section not found.');
+    }
+  } else {
+    throw new Error("'## [Unreleased]' not found in the file.");
```

Comment:
I think we will have a scenario where `[Unreleased]` will not exist, perhaps when patching a released branch or even due to human error. I believe this should not fail but gracefully handle it.

Reply:
@AMoo-Miki, the part for the release notes part was done by other contributors. I need to check their code with calm to do the updates you suggest. For this specific purpose you are saying we should use a `try/catch` block?

Reply:
Hey @BigSamu working on this right now. Should have a commit in the next few minutes.

Reply:
Great @CMDWillYang! Many thanks. I tagged you in other reviews that @AMoo-Miki sent for the release notes part. I would really appreciate it if you could review them.

Reply:
@AMoo-Miki many thanks for the review! This was initially added because @ashwin-pc suggested that not having unreleased section should never happen. Are you suggesting that we manually add that header and not terminate the script?

https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5519#discussion_r1409211443
<img width="811" alt="Screenshot 2024-04-11 at 11 12 13 PM" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/70486227/e3577fb7-51dc-436e-934c-670b43cc3d3c">